### ISSUE TITLE
Allow PeerAuthentication on ingress gateways

### ISSUE THREAD

**Issue by chaudyg** [istio/istio#23284]
**Describe the feature request**
Allow Peer authentication (aka Istio mTLS) to work on gateway.

Currently, you can set TLS/mTLS via gateway API. This however doesn't add the authn filter on the gateway, which is confusing. For more details, see https://github.com/istio/istio/issues/21343.

**Describe alternatives you've considered**

The following envoy filter is a an alternative workaround.
```
---
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: authn
  namespace: istio-system
spec:
  configPatches:
  - applyTo: HTTP_FILTER
    match:
      context: GATEWAY
      listener:
        portNumber: 80
        filterChain:
          filter:
            name: "envoy.http_connection_manager"
            subFilter:
              name: "envoy.filters.http.rbac"
    patch:
      operation: INSERT_BEFORE
      value:
        name: "istio_authn"
        config:
          "@type": "type.googleapis.com/istio.envoy.config.filter.http.authn.v2alpha1.FilterConfig"
          policy:
            peers:
            - mtls: {}
```

**Additional context**
For more details, see https://github.com/istio/istio/issues/21343

**Comment by JimmyCYJ** [comment]
@yangminzhu can I assign to you for triage?

**Comment by howardjohn** [comment]
This has been fixed by #25818 cc @rshriram

**Comment by chaudyg** [comment]
Thanks @howardjohn 🎉 . I will try it out !
